src/dilithium.cpp

```diff
@@ -0,0 +1,448 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2022 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include "mococrw/dilithium.h"
+#include <boost/format.hpp>
+#include <boost/format/format_fwd.hpp>
+#include <cstdint>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+#include <vector>
+#include "mococrw/asymmetric_crypto_ctx.h"
+#include "mococrw/error.h"
+#include "mococrw/key.h"
+#include "mococrw/openssl_wrap.h"
+
+extern "C" {
+#include <dilithium-3.1/api.h>
+#include <openssl/asn1.h>
+#include <openssl/asn1t.h>
+#include <openssl/err.h>
+#include <openssl/objects.h>
+#include <openssl/x509.h>
+}
+
+namespace mococrw
+{
+using KeyTypes = AsymmetricKey::KeyTypes;
+
+/*
+ * The dilithium private key struct as returned by the backend
+ * (iaik: http://javadoc.iaik.tugraz.at/iaik_jce/current/iaik/pkcs/pkcs8/PrivateKeyInfo.html)
+ *
+ * This is the internal struct inside the OCTET_STRING
```

Comment:
Can you refer to the attributes by name instead of their type? 
```suggestion
 * This is the internal structure contained in the PrivateKey element of the PrivateKeyInfo structure.
```